src/Symfony/Component/Form/Extension/Csrf/EventListener/CsrfValidationListener.php

```diff
@@ -66,7 +67,8 @@ public function preSubmit(FormEvent $event)
                     $errorMessage = $this->translator->trans($errorMessage, array(), $this->translationDomain);
                 }
 
-                $form->addError(new FormError($errorMessage));
+                $cause = new ConstraintViolation($errorMessage, null, array(), null, '', null, null, 'csrf');
+                $form->addError(new FormError($errorMessage, null, array(), null, $cause));
```

Comment:
Using a `ConstraintViolation` here adds a hard dependency on the Validator component. This is not necessary. A cause of a form error can be arbitrary objects. We should add a new class for that.

Reply:
I hear what you are saying but I am a little bit confused.

If I understood correctly you are suggesting this:

```
$cause = new CsrfValidationError($some_arguments_or_no_arguments_at_all);
$form->addError(new FormError($errorMessage, null, array(), null, $cause));
```

The reason I went with my implementation is because of this [line](https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Form/FormErrorIterator.php#L268)

The function above implies that all causes are `ConstraintViolation` objects. Personally I can fulfill my requirements with your suggestion but IMHO I think 

- On this current PR the cause should be a simple string.
- `$cause` argument on `FormError` object should either accept strings objects that implement the `__toString` function or a `ConstraintViolation` object.
- Besides filtering `ConstraintViolations` the function `findByCodes` should also be able to filter `causes` that are string or objects that implement `__toString` function.

If you still believe that what you are saying is the way to go I will change my PR accordingly. Let me know.

Reply:
btw I do not suggest to implement all these changes at the current PR.

Reply:
Hello, any feedback on this... sorry if I am pushing things.

Reply:
I believe I have applied the changes :)